### PR TITLE
Fixes #19213: Put node cache info into a file in place of LDAP

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/19213

### CODE REVIEW

**Comment by fanf** [comment]
PR updated with a new commit

**Comment by ncharles** [comment]
something is not working as expected, I get a non existent file after a policy generaiton, and during I have a 
{"hashes": [
  
] }

**Comment by fanf** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Comment by fanf** [comment]
OK, squash merging this PR

File: webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala [hunk 1]
```diff
@@ -390,6 +432,125 @@ class InMemoryNodeConfigurationHashRepository extends NodeConfigurationHashRepos
...
+      hashes <- NodeConfigurationHashes.fromJson(json)
+    } yield hashes).catchAll(err =>
+      PolicyGenerationLoggerPure.error(s"Error when trying to read node configuration hashes, they will be ignored: a full generation will take place. Error was: ${err.fullMsg}") *>
+      NodeConfigurationHashes(Nil).succeed
+    )
+  }
+
+  private def nonAtomicWrite(hashes: NodeConfigurationHashes): IOResult[Unit] = {
+    import java.nio.file.StandardOpenOption._
+    IOResult.effect(hashesFile.writeText(NodeConfigurationHashes.toJson(hashes))(Seq(WRITE, CREATE, TRUNCATE_EXISTING), StandardCharsets.UTF_8))
```
Comment (ncharles) [inline webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/nodeconfig/NodeConfigurationCacheRepository.scala:507]: this replace the whole content of the file with the hashes

**Pull request merged**